File: IT/2/Input-Output/Esercizi/Java/Esercizio7.java
/* file Esercizio7.java */

import java.io.RandomAccessFile;
import java.io.FileNotFoundException;
import java.io.IOException;

public class Esercizio7 {
	public static void main(String [] args) {
		String filename = System.console().readLine("Apri il file: ");

		try {
			RandomAccessFile f = new RandomAccessFile(filename, "r");

			while (f.getFilePointer() < f.length()) {
				System.out.println(f.readLine());
			}

			System.out.println("-> torna alla prima riga.");

			f.seek(0);

			while (f.getFilePointer() < f.length()) {
				System.out.println(f.readLine());
			}

			f.close();
		}
		catch (FileNotFoundException e) {
			System.out.println(e);
		}
		catch (IOException e) {
			e.printStackTrace();
		}
	}
}

File: IT/2/Input-Output/Esercizi/C++/esercizio7.cpp
/*
file esercizio7.cpp
compilare con: g++ -std=c++11 esercizio7.cpp
*/

#include <iostream>
#include <fstream>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::ifstream;

int main()
{
	cout << "Apri il file: ";

	string filename;
	getline(cin, filename);

	ifstream f(filename);
	string s;

	while (getline(f, s)) {
		cout << s << endl;
	}

	cout << "-> torna alla prima riga." << endl;

	f.clear();
	f.seekg(0, ifstream::beg);

	while (getline(f, s)) {
		cout << s << endl;
	}
}

